spec/migrations/20180823111741_move_location_led_state_to_asset_details_table_spec.rb

```diff
@@ -0,0 +1,85 @@
+require_migration
+
+describe MoveLocationLedStateToAssetDetailsTable do
+  let(:physical_server_stub)  { migration_stub(:PhysicalServer) }
+  let(:physical_chassis_stub) { migration_stub(:PhysicalChassis) }
+  let(:asset_detail_stub)     { migration_stub(:AssetDetail) }
+
+  context "resources with asset detail" do
+    migration_context :up do
+      it "should migrate the data from physical_chassis and physical_servers to asset_details table" do
+        asset_detail_server = asset_detail_stub.create!
+        physical_server_stub.create!(
+          :asset_detail       => asset_detail_server,
+          :location_led_state => 'Off'
+        )
+
+        asset_detail_chassis = asset_detail_stub.create!
+        physical_chassis_stub.create!(
+          :asset_detail       => asset_detail_chassis,
+          :location_led_state => 'Blinking'
+        )
+
+        migrate
+
+        expect(asset_detail_server.reload.location_led_state).to  eq('Off')
+        expect(asset_detail_chassis.reload.location_led_state).to eq('Blinking')
+      end
+    end
+
+    migration_context :down do
+      it "should migrate the data back from asset_details to physical_chassis and physical_servers tables" do
+        asset_detail_server = asset_detail_stub.create!(
+          :location_led_state => 'Off'
+        )
+        server = physical_server_stub.create!(
+          :asset_detail => asset_detail_server
+        )
+
+        asset_detail_chassis = asset_detail_stub.create!(
+          :location_led_state => 'Blinking'
+        )
+        chassis = physical_chassis_stub.create!(
+          :asset_detail => asset_detail_chassis
+        )
+
+        migrate
+
+        expect(server.reload.location_led_state).to  eq('Off')
+        expect(chassis.reload.location_led_state).to eq('Blinking')
+      end
+    end
+  end
+
+  context "resources without asset detail" do
+    migration_context :up do
+      it "should migrate the data from physical_chassis and physical_servers to asset_details table" do
+        server = physical_server_stub.create!(
+          :location_led_state => 'Off'
+        )
+
+        chassis = physical_chassis_stub.create!(
+          :location_led_state => 'Blinking'
+        )
+
+        migrate
+
+        expect(server.reload.asset_detail.location_led_state).to  eq('Off')
```

Comment:
Can you verify that the `asset_detail.resource_type` is `"PhysicalServer"` not `"MoveLocationLedStateToAssetDetailsTable::PhysicalServer"`

Reply:
But, it is `MoveLocationLedStateToAssetDetailsTable::PhysicalServer`, once I'm using the class defined inside `MoveLocationLedStateToAssetDetailsTable`. I thought that it is the expected result :thinking:. Anyway, is there something that I'm missing here to make the asset_detail.resource_type point to "PhysicalServer" ?

Reply:
The problem is that `MoveLocationLedStateToAssetDetailsTable::PhysicalServer` only exists during migration.  For production use, only `PhysicalServer` will exist.

Reply:
I'm not totally familiarized with the schema tests environment, please, correct me if I'm wrong. As far as I know, the tests suite doesn't has access to ManageIQ models, so we need to define those models like I'm doing in [here](https://github.com/ManageIQ/manageiq-schema/pull/262/files#diff-a18ede530fc209a9f9daed548de9f3b1R8), so when I create the relationship between server and asset_detail inside the specs the `asset_detail.resource_type` will be `MoveLocationLedStateToAssetDetailsTable::PhysicalServer`. Anyway, if I run this migration, everything works fine, once it is changing the right tables and tuples.

If you insist that I must do this test, I will need to configure some more thing to make `asset_detail.resource_type` point to `"PhysicalServer"`, could you tell me what is this configuration that I'm missing here?

Reply:
Here's how I would write it to avoid the incorrect class names... https://github.com/douglasgabriel/manageiq-schema/pull/1